### PR TITLE
Setup db

### PR DESCRIPTION
**Pull request by tchiu2**
Added Postgres and Sequelize config. To run locally:
1. Start Postgres
2. Update ```DATABASE_USER``` and ```DATABASE_PASSWORD``` in ```.env``` file
3. Run ```npm run db:setup``` in root directory.

### CODE REVIEW

File: server/models/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+const fs = require('fs');
+const path = require('path');
+const Sequelize = require('sequelize');
+const basename = path.basename(__filename);
+const env = process.env.NODE_ENV || 'development';
+const config = require(__dirname + '/../config/config.js')[env];
+const db = {};
+
+let sequelize;
```
Comment (achiu8) [inline server/models/index.js]: Nit: let's use ternary.

File: server/models/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+if (config.use_env_variable) {
+  sequelize = new Sequelize(process.env[config.use_env_variable], config);
+} else {
+  sequelize = new Sequelize(config.database, config.username, config.password, config);
+}
+
+fs
```
Comment (achiu8) [inline server/models/index.js]: The syntax in this file is kinda funny--did you copy it from somewhere? We can clean it up:
```js
const db = fs
  .readdirSync(__dirname)
  .filter(file => file.indexOf('.') !== 0 && file !== basename && file.slice(-3) === '.js')
  .reduce((acc, file) => {
    const model = sequelize.import(path.join(__dirname, file));
    return {
      ...acc,
      [model.name]: model
    };
  }, { sequelize, Sequelize });
```

File: server/routes/index.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+const express = require('express');
+const router = express.Router();
+
+const userController = require('../controllers').user;
+
+router.get('/api/user', userController.list);
```
Comment (achiu8) [inline server/routes/index.js]: If list, should this be `/api/users`?

File: server/server.js [hunk 4]
```diff
@@ -2,11 +2,13 @@ require('dotenv').config()
...
 const google = require('./routes/google');
 const dropbox = require('./routes/dropbox');
+const indexRouter = require('./routes/index');
 
 const app = express();
 const port = process.env.PORT || 3001;
 
 app.use('/google', google);
 app.use('/dropbox', dropbox);
+app.use('/', indexRouter);
```
Comment (achiu8) [inline server/server.js]: Maybe we nest this under `/api` and call it `apiRouter`? That way, we won't have to be careful about order here, or prefix all the API routes.

File: server/config/config.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+const fs = require('fs');
```
[the remark this answers is not in the document]
Comment (tchiu2) [reply server/config/config.js]: Ah, must have missed that. Took the example from [Sequelilze Migration docs](http://docs.sequelizejs.com/manual/migrations.html#production-usages) which looks like ```fs``` is used to read the SSL certs for production.

Will remove.

File: server/config/config.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+module.exports = {
+  "development": {
+    "username": process.env.DATABASE_USER,
+    "password": process.env.DATABASE_PASSWORD,
+    "database": "atlas-development",
```
[the remark this answers is not in the document]
Comment (tchiu2) [reply server/config/config.js]: Will change.

File: server/controllers/user.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+module.exports = {
+  list(req, res) {
+    return User
+      .findAll({
+        order: [
+          ['createdAt', 'DESC'],
+        ],
+      })
+      .then(users => res.status(200).send(users))
+      .catch(err => { res.status(404).send(err); });
```
[the remark this answers is not in the document]
Comment (tchiu2) [reply server/controllers/user.js]: Will remove.

File: server/migrations/20190330223020-create-user.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
+'use strict';
```
[the remark this answers is not in the document]
Comment (tchiu2) [reply server/migrations/20190330223020-create-user.js]: I think this is a default when using ```sequelize-cli``` to generate migrations (i.e. ```npx sequelize model:generate --name User --attributes firstName:string,lastName:string,email:string```).

I don't feel strongly about it either way, so what are your thoughts on using it (or not)?

File: server/migrations/20190330223020-create-user.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('Users', {
```
[the remark this answers is not in the document]
Comment (tchiu2) [reply server/migrations/20190330223020-create-user.js]: Also appears to be the default behavior when using ```sequelize-cli``` to generate migrations.

Will change and keep this in mind for future migrations.

File: server/models/index.js (hunk 1 above)
Comment (tchiu2) [reply server/models/index.js]: Sorry, not quite seeing where to use the ternary. Could you clarify?

File: server/models/index.js (hunk 2 above)
Comment (tchiu2) [reply server/models/index.js]: Also, generated by ```sequelize-cli```. I agree and like your version. Will update.

File: server/routes/index.js (hunk 3 above)
Comment (tchiu2) [reply server/routes/index.js]: Yup, agreed. Will change.

File: server/server.js (hunk 4 above)
Comment (tchiu2) [reply server/server.js]: That makes sense. Will update.

File: server/migrations/20190330223020-create-user.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (achiu8) [reply server/migrations/20190330223020-create-user.js]: Oh okay--if it's automatically generated, let's just leave them / not worry about adding elsewhere.

File: server/migrations/20190330223020-create-user.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (achiu8) [reply server/migrations/20190330223020-create-user.js]: Yeah kind of annoying that they violate some pretty common naming conventions by default.

File: server/models/index.js [hunk 10]
```diff
@@ -6,32 +6,23 @@ const Sequelize = require('sequelize');
 const basename = path.basename(__filename);
 const env = process.env.NODE_ENV || 'development';
 const config = require(__dirname + '/../config/config.js')[env];
-const db = {};
```
Comment (tchiu2) [inline server/models/index.js]: Not sure how you intended to use the ternary here, but definitely let me know if there's a more conventional/preferred way to do it.

Comment (achiu8) [reply server/models/index.js]: Oh could you just do this?
```js
const sequelize = config.use_env_variable
  ? new Sequelize(process.env[config.use_env_variable], config)
  : new Sequelize(config.database, config.username, config.password, config);
```

Comment (tchiu2) [reply server/models/index.js]: Cool, that was actually how I initially thought to write it - should've just went with my gut.

Updated.

**Pull request merged**